src/SelectField.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+/**
+ * Represents a Select Field.
+ * @license GNU GPL v2+
+ * @since 3.0.0
+ * @author: Alexander Gesinn
+ */
+
+namespace SFS;
+
+use SMWQueryProcessor as QueryProcessor;
+use Parser;
+use MWDebug;
+
+class SelectField {
+
+	private $mParser = null;
+
+	//private $mSelectField = array();
+	private $mValues = null;
+	private $mHasStaticValues = false;
+
+	private $mData = array();    # array with all parameters
+	private $mQuery = "";
+	private $mFunction = "";
+	private $mSelectIsMultiple = false;
+	private $mSelectTemplate = "";
+	private $mSelectField = "";
+	private $mValueTemplate = "";
+	private $mValueField = "";
+	private $mSelectRemove = false;
+	private $mLabel = false;
+	private $mDelimiter = ",";
+
+	public function __construct( & $parser ) {
+		$this->mParser = $parser;
+	}
+
+	/**
+	 * Convenience function to process all parameters at once
+	 */
+	public function processParameters( $input_name = "", $other_args ) {
+		if ( array_key_exists( "query", $other_args ) ) {
+			$this->setQuery( $other_args );
+		} elseif ( array_key_exists( "function", $other_args ) ) {
+			$this->setFunction( $other_args );
+		}
+	}
+
+	/**
+	 * getData
+	 *
+	 * @return array Array with all parameters
+	 */
+	public function getData() {
+		return $this->mData;
+	}
+
+	public function setQuery( $other_args ) {
+		$query = $other_args["query"];
+		$query = str_replace( array( "~", "(", ")" ), array( "=", "[", "]" ), $query );
+
+		//$this->mSelectField["query"] = $query;
+		$this->mQuery = $query;
+		$this->mData['selectquery'] = $query;
+
+		// unparametrized query
+		if ( strpos( $query, '@@@@' ) === false ) {
+			$params = explode( ";", $query );
+
+			// there is no need to run the parser, $query has been parsed already
+			//$params[0] = $wgParser->replaceVariables( $params[0] );
+
+			$this->mValues = QueryProcessor::getResultFromFunctionParams( $params, SMW_OUTPUT_WIKI );
+
+			$this->setHasStaticValues( true );
+		}
+	}
+
+	public function setFunction( $other_args ) {
+		#global $wgParser;
+
+		$function = $other_args["function"];
+		$function = '{{#' . $function . '}}';
+		$function = str_replace( array( "~", "(", ")" ), array( "=", "[", "]" ), $function );
+
+		//$this->mSelectField["function"] = $function;
+		$this->mFunction = $function;
+		$this->mData['selectfunction'] = $function;
+
+		// unparametrized function
+		if ( strpos( $function, '@@@@' ) === false ) {
+			$f = str_replace( ";", "|", $function );
+
+			$this->setValues( $this->mParser->replaceVariables( $f ) );
+
+			$this->setHasStaticValues( true );
+		}
+	}
+
+	/**
+	 * setSelectIsMultiple
+	 * @param string[] $other_args
+	 */
+	public function setSelectIsMultiple( Array $other_args ) {
+		$this->mSelectIsMultiple = array_key_exists( "part_of_multiple", $other_args );
+		$this->mData["selectismultiple"] = $this->mSelectIsMultiple;
+	}
+
+	/**
+	 * setSelectTemplate
+	 * @param string $input_name
+	 */
+	public function setSelectTemplate( $input_name = "" ) {

```

Comment:
These descriptions are really not helpful :) Typically no description is warranted for sane methods, so just don't add any.